pages/profile/collections/[id].js

```diff
@@ -0,0 +1,38 @@
+import IconButton from "@/components/Styled/IconButton";
+import MealCard from "@/components/MealCard";
+import { useRouter } from "next/router";
+import StyledUl from "@/components/StyledUl";
+import Spacer from "@/components/Styled/Spacer";
+import StyledH2 from "@/components/Styled/StyledH2";
+
+export default function DetailCollection({ recipes, user }) {
+  const router = useRouter();
+  const { id } = router.query;
+  if (!user || !recipes) {
+    return;
+  }
+
+  const foundCollection = user.collections.find((col) => col._id === id);
+
+  const foundRecipes = recipes.filter((recipe) =>
+    foundCollection.recipes.includes(recipe._id)
```

Comment:
Can you use the populate-function in the user API-route instead and access the recipes directly?